app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(in_func: Callable) -> Callable:
+    result_dict = {}
+
+    def inner(*args, **kwargs) -> Callable:
```

Comment:
Are you sure that it returns Callable?)

Reply:
You are right, I don't really know what type it returns because it returns result of some function that will be cached. So it's Any